src/transformers/models/mamba/modeling_mamba.py

```diff
@@ -633,18 +633,30 @@ def _update_model_kwargs_for_generation(
         return model_kwargs
 
     def prepare_inputs_for_generation(
-        self, input_ids, cache_params: Optional[MambaCache] = None, inputs_embeds=None, attention_mask=None, **kwargs
+        self,
+        input_ids,
+        inputs_embeds=None,
+        attention_mask=None,
+        use_cache=None,
+        past_key_values: Optional[MambaCache] = None,
+        **kwargs,
     ):
         # only last token for inputs_ids if the state is passed along.
-        if cache_params is not None:
+        if past_key_values is not None:
```

Comment:
I wouldn't rename the cache params to past kvs since mamba is not based on the attention mechanism making this a bit confusing. (The cache in this case saves past outputs of the ssm and conv1d layers.)

It's also a bit confusing in the generation utils but is kinda a necessity due to the code structure.

Reply:
Yes, I agree here but this is done for consistency, and following other models that use a different name for cache (like xlnet). 

I will leave this to @gante if we want to use the same naming for cache or not, as we're in the stage of refactoring generate :)

Reply:
➕ on @vasqu , the older models like XLNET are not good precedents. 
Let's keep `cache_params`